src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotHaveIncorrectSuffix.cs

```diff
@@ -147,12 +149,17 @@ public override void Initialize(AnalysisContext analysisContext)
                                 return;
                             }
 
+                            var allowedSuffixes = namedTypeSymbol.DeclaringSyntaxReferences
+                                .Select(syntaxRef => symbolAnalysisContext.Options.GetStringOptionValue(EditorConfigOptionNames.AllowedSuffixes, TypeNoAlternateRule, syntaxRef.SyntaxTree, symbolAnalysisContext.Compilation, symbolAnalysisContext.CancellationToken))
```

Comment:
We might have multiple syntax-trees and I wasn't sure how to best handle it so I decided to merge the various results. Let me know if you prefer something else.

Reply:
Lets just stick to the first tree, see https://github.com/dotnet/roslyn-analyzers/issues/3494 for context.

Also its cheaper to use `symbol.Locations[0].SyntaxTree` instead of `DeclaringSyntaxReferences`, I think the latter does not cache things and/or has more allocations.